src/modules/pos/reward/hooks/queries/useRewardsClaimable.js

```diff
@@ -20,21 +21,43 @@ import { useCustomQuery } from '@common/hooks';
  * @returns the query object
  */
 
-export const useRewardsClaimable = ({ config: customConfig = {}, options } = { }) => {
-  const hasRequiredParams = customConfig.params?.address || customConfig.params?.name || customConfig.params?.publicKey;
+export const useRewardsClaimable = ({ config: customConfig = {}, options } = {}) => {
+  const hasRequiredParams =
+    customConfig.params?.address || customConfig.params?.name || customConfig.params?.publicKey;
+
+  const createMetaConfig = useAppsMetaTokensConfig();
+  const transformToken = addTokensMetaData({ createMetaConfig, client: defaultClient });
+
+  const transformResult = async (res) => {
+    const tokens = await transformToken(res.data);
+    const rewardsWithToken = res.data?.map((rewardsClaimable) => {
+      const token = tokens.find((metaToken) => metaToken.tokenID === rewardsClaimable.tokenID);
+
+      return {
+        ...rewardsClaimable,
+        ...token,
+      };
+    });
+    return {
+      ...res,
+      data: rewardsWithToken,
+    };
+  };
```

Comment:
This would result to a Big O of n^2 (Quadratic time)

Instead can't we convert tokens to an object whose key it the tokenID and value is the token then within the `.map` on rewards response, we can just reference the token by accessing the token from the token map object. I think something like this

```suggestion
    const mapIDToToken = tokens.reduce((result, token) => ({...result, [token.tokenID] : token}) , {})
    const rewardsWithToken = res.data?.map((rewardsClaimable) => {
      const token = mapIDToToken[rewardsClaimable.tokenID]

      return {
        ...rewardsClaimable,
        ...token,
      };
    });
    return {
      ...res,
      data: rewardsWithToken,
    };
  };
```

This way we can reduce this logic's time complexity from Quadratic time to linear time

Reply:
Doing tokens.reduce and than merge has its own performance implications. Was curious about this so i ran it in a perf bench tool, seems to be 86% slower doing it like that. 
Below are some fun tools one can experiment with to test the performance of different operations 
https://jsben.ch/1M9pv
https://jsperf.app/voloyi

Reply:
instead of a reduce we can just use a simple for operation to generate the map. eg

```suggestion
    const mapIDToToken = {};
    tokens.forEach((token) => { 
      mapIDToToken[token.tokenID] = token 
    })
    const rewardsWithToken = res.data?.map((rewardsClaimable) => {
      const token = mapIDToToken[rewardsClaimable.tokenID]
      return {
        ...rewardsClaimable,
        ...token,
      };
    });
    return {
      ...res,
      data: rewardsWithToken,
    };
  };
```

my suggestion actually is that, I think we can do linear time instead of a quadratic time

Reply:
If you paste it in the above tools you will find that this is still 80% slower